Backend/Controllers/ProductController.cs

```diff
@@ -0,0 +1,252 @@
+           
+
+            if (patchDoc.SeoDescription != null) product.SeoDescription = patchDoc.SeoDescription;
+            
+
+            var saved = await db.Save();
+
+            return saved ? Ok(product) : Problem();
+        }
+
+
+        [HttpPut("{id}/publish")]
+        public async Task<IActionResult> PublishProduct(string id)
+        {
+            var product = await db.Products.QueryOne(p => p.Id == id);
+
+            if (product == null) return Problem();
+
+            var uid = User.FindFirst(Jwt.Uid)?.Value;
+            var shop = await db.Shops.QueryOne(s => s.Id == product.ShopId && s.OwnerId == uid);
+           
+            if (shop == null) return Forbid();
+
+            product.IsDraft = false;
+            var saved = await db.Save();
+
+            return saved ? Ok() : Problem();
+        }
+
+
+        [HttpPost("{id}/image")]
+        public async Task<IActionResult> UploadProductImage(string id, [FromForm] IFormFile imageFile)
+        {
+                var product = await db.Products.QueryOne(p => p.Id == id);
+
+                if (product == null) return Problem();
+
+                var uid = User.FindFirst(Jwt.Uid)?.Value;
+                var shop = await db.Shops.QueryOne(s => s.Id == product.ShopId && s.OwnerId == uid);
+
+                if (shop == null) return Forbid();
+
+                if (!IsImageFile(imageFile)) return BadRequest();
+
+                var fileId = Guid.NewGuid().ToString() + Path.GetExtension(imageFile.FileName);
+
+                using (var stream = imageFile.OpenReadStream())
+                {
+                   var uploadedFile = await storage.Upload(fileId, stream);
+
+                   if (uploadedFile == null) return Problem();
+                   
+                   var productImage = new ProductImage(uploadedFile.Provider, uploadedFile.Bucket, uploadedFile.Key, id);
+                   db.ProductImages.Add(productImage);
+                   
+                }
+
+                var saved = await db.Save();
+                return saved ? Ok() : Problem();
+        }
+
+        [NonAction]
+        public static bool IsImageFile(IFormFile file)
+        {
+            if (file == null || string.IsNullOrEmpty(file.FileName) || file.Length == 0)
+            {
+                return false;
+            }
+
+            var fileExtension = Path.GetExtension(file.FileName).ToLower();
+            string[] photoExtensions = { ".jpg", ".jpeg", ".png", ".gif", ".bmp", ".svg", ".webp", ".tiff", ".ico", ".jfif", ".psd", ".eps", ".pict", ".pic", "pct" }; ;
+
+            return photoExtensions.Contains(fileExtension);
+        }
+
+        [HttpDelete("{id}/image")]
+        public async Task<IActionResult> DeleteProductImage(string fileId)
+        {
+            var productImagesToDelete = await db.ProductImages.QueryOne(pi => pi.Id == fileId);
+
+            var uid = User.FindFirst(Jwt.Uid)?.Value;
```

Comment:
no! no! no! wtf. 2 queries for what. do it in 1 query. get shop id from path!!! 
and